File: GiaiThuat_Tim_BCNN.java
// Yêu cầu:
// 1. nhập a
// 2. nhập b
// 3. nhập c
// 4. nhập d

// 5. tìm bội chung nhỏ nhất BCNN(a, b, c, d)

// VD: BCNN(2, 3, 4, 8) = 24

import java.util.Scanner;

class Program{
    // tạo hàm max(int, int, int, int) có 4 tham số
    public static int max(int thamSo1, int thamSo2, int thamSo3, int thamSo4){
        int max = thamSo1;

        // tạo mảng để tìm max
        int[] arr = {thamSo1, thamSo2, thamSo3, thamSo4};

        for (int item : arr) {
            if (max < item) {
                max = item;
            }
        }

        return max;
    }

    // tạo hàm tìm bội chung nhỏ nhất
    public static int Ham_Tim_BCNN(int thamSo1, int thamSo2, int thamSo3, int thamSo4){
        int BCNN = 0;

        // tạo index bắt đầu
        int index_bat_dau = max(thamSo1, thamSo2, thamSo3, thamSo4);

        // tạo index kết thúc
        int index_ket_thuc = thamSo1 * thamSo2 * thamSo3 * thamSo4;

        for (int i = index_bat_dau; i <= index_ket_thuc; i++) {
            boolean kiemTra1 = i % thamSo1 == 0? true : false;
            boolean kiemTra2 = i % thamSo2 == 0? true : false;
            boolean kiemTra3 = i % thamSo3 == 0? true : false;
            boolean kiemTra4 = i % thamSo4 == 0? true : false;

            if (kiemTra1 && kiemTra2 && kiemTra3 && kiemTra4) {
                BCNN = i;

                // tìm được bội chung nhỏ nhất rồi thì phải break
                break;          
            }
        }

        return BCNN;
    }

    public static void main(String[] agrs){
        // tạo đối tượng đọc dữ liệu nhập từ bàn phím
        Scanner dt = new Scanner(System.in);

        int a = 0;
        int b = 0;
        int c = 0;
        int d = 0;

        System.out.print("Nhap vao a: ");
        a = dt.nextInt();
        
        System.out.print("Nhap vao b: ");
        b = dt.nextInt();
        
        System.out.print("Nhap vao c: ");
        c = dt.nextInt();
        
        System.out.print("Nhap vao d: ");
        d = dt.nextInt();

        // gọi hàm tìm bội chung nhỏ nhất
        int x = Ham_Tim_BCNN(a, b, c, d);

        System.out.print("\nBCNN(" + a + ", " + b + ", " + c + ", " + d + ") = " + x);
    }
}

File: GiaiThuat_Tim_BCNN.py
# Yêu cầu:
# 1. nhập vào a
# 2. nhập vào b
# 3. nhập vào c
# 4. nhập vào d

# 5. tìm bội chung nhỏ nhất của a, b, c, d

# ký hiệu BCNN(a, b, c, d)

a = int(input("Nhap vao a: "))
b = int(input("Nhap vao b: "))
c = int(input("Nhap vao c: "))
d = int(input("Nhap vao d: "))

# hàm tìm bội chung nhỏ nhất
def Ham_Tim_BCNN(thamSo1, thamSo2, thamSo3, thamSo4):
    BoiChungNhoNhat = 0
    
    index_bat_dau = max(thamSo1, thamSo2, thamSo3, thamSo4)

    index_ket_thuc = thamSo1 * thamSo2 * thamSo3 * thamSo4

    for i in range (index_bat_dau, index_ket_thuc + 1, 1):
        kiemTra1 = True if i % thamSo1 == 0 else False
        kiemTra2 = True if i % thamSo2 == 0 else False
        kiemTra3 = True if i % thamSo3 == 0 else False
        kiemTra4 = True if i % thamSo4 == 0 else False

        if kiemTra1 and kiemTra2 and kiemTra3 and kiemTra4:
            BoiChungNhoNhat = i
            break

    return BoiChungNhoNhat

# gọi hàm tìm bội chung nhỏ nhất
x = Ham_Tim_BCNN(a, b, c, d)

print(f"BCNN({a}, {b}, {c}, {d}) = {x}")